src/v/redpanda/admin_server.cc

```diff
@@ -588,13 +591,31 @@ ss::future<> admin_server::configure_listeners() {
 
 void admin_server::log_request(
   const ss::http::request& req, const request_auth_result& auth_state) const {
-    vlog(
-      logger.debug,
-      "[{}] {} {}",
-      auth_state.get_username().size() > 0 ? auth_state.get_username()
-                                           : "_anonymous",
-      req._method,
-      req.get_url());
+    const auto username = auth_state.get_username().size() > 0
+                            ? auth_state.get_username()
+                            : "_anonymous";
+    vlog(logger.debug, "[{}] {} {}", username, req._method, req.get_url());
+
+    bool pass
+      = _audit_mgr.local().enqueue_audit_event<security::audit::api_activity>(
+        security::audit::event_type::management,
+        make_api_activity_event(
+          req,
+          security::audit::authorization_allowed{
+            .username = username,
+            .mechanism = auth_state.get_sasl_mechanism(),
+            .is_superuser = auth_state.is_superuser()}));
+
+    if (!pass) {
+        /// Failed to audit (when audit is enabled) results in a 5xx response.
```

Comment:
500 is probably not what we want for all scenarios. what do you want the client to do in response?

should `pass` be called something else if it represents "full audit queue"? or am i misunderstanding the meaning of "pass" in this context?

Reply:
Surely a [503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) is better? 500 should be only for unexpected error conditions.

Reply:
sure